### PR TITLE
DATA: User Model and Migration

### PR DESCRIPTION
**Pull request by MahalakshmiMurari**

### CODE REVIEW

**Comment by MahalakshmiMurari** [comment]
image

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            var serviceAdminUser = new User
+            {
+                //Id = Guid.Parse("80aea9ae-967a-4718-b04e-97d2614b862b")
+                Id = Guid.Parse("4c54b038-c653-4904-a33f-54e60ca4d531"),
+                Name = "Admin"
+            };
+
+            modelBuilder.Entity<User>().HasData(serviceAdminUser);
+
```
Comment (NaynaMallick) [inline ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs]: remove extra line

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs [hunk 2]
```diff
@@ -18,9 +18,13 @@ public StorageBroker(IConfiguration configuration)
             this.Database.Migrate();
         }
 
+        protected override void OnModelCreating(ModelBuilder modelBuilder) =>
+            SeedUsers(modelBuilder);
+
```
Comment (NaynaMallick) [inline ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs]: remove extra line

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+namespace ReserveSpot.Services.Api.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        private static void SeedUsers(ModelBuilder modelBuilder)
+        {
+            var serviceAdminUser = new User
+            {
+                //Id = Guid.Parse("80aea9ae-967a-4718-b04e-97d2614b862b")
```
Comment (ShriHumrudha) [inline ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs]: remove commented code

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs (hunk 1 above)
Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs]: Removed extra line

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs (hunk 2 above)
Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs]: Removed extra line

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs (hunk 3 above)
Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs]: Removed commented code

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        private static void SeedUsers(ModelBuilder modelBuilder)
+        {
+            var serviceAdminUser = new User
+            {
+                Id = Guid.Parse("4c54b038-c653-4904-a33f-54e60ca4d531"),
+                Name = "Admin"
+            };
+
+            modelBuilder.Entity<User>().HasData(serviceAdminUser);
+
```
Comment (NaynaMallick) [inline ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs]: remove extra line 22

Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.User.Seeds.cs]: Removed extra line

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs (hunk 2 above)
Comment (ShriHumrudha) [reply ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs]: I still see 23 and 24 as empty lines

File: ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj [hunk 5]
```diff
@@ -7,13 +7,15 @@
 
 	<ItemGroup>
 		<PackageReference Include="EFxceptions" Version="0.4.5" />
+		 <PackageReference Include="Microsoft.AspNetCore.OData" Version="8.0.8" />
```
Comment (ShriHumrudha) [inline ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj]: Why do we need OData?

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs (hunk 2 above)
Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/Brokers/Storages/StorageBroker.cs]: Removed extra line 24

File: ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj (hunk 5 above)
Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj]: This defines a composite OData query options that can be used to perform query composition. Right now, it is not required. By mistake it was copied from another file. So deleted that entry.

File: ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj [hunk 6]
```diff
@@ -7,13 +7,14 @@
 
 	<ItemGroup>
 		<PackageReference Include="EFxceptions" Version="0.4.5" />
-		<PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.2" />
+	    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.2" />
```
Comment (ShriHumrudha) [inline ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj]: fix alignment

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj (hunk 6 above)
Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/ReserveSpot.Services.Api.csproj]: fixed alignment

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**